workflows/hic/hic-post.wdl

```diff
@@ -0,0 +1,104 @@
+version 1.1
+
+import "../../tools/hilow.wdl"
+import "../../tools/picard.wdl"
+import "../../tools/samtools.wdl"
+
+workflow hic_post {
+    meta {
+        description: "Post-processing of Hi-C data."
+        outputs: {
+            hic_file: "Interactions in .hic format",
+            filtered_pairs: "File with filtered list of interaction pairs",
+            qc_report: "QC report for Hi-C experiment",
+            combined_bam: "Merged and duplicate marked BAM file",
+            combined_bam_index: "Index for the combined BAM file",
+        }
+    }
+
+    parameter_meta {
+        all_valid_pairs: "All valid pairs file from HiC-Pro"
+        chromsizes: {
+            description: "Tab delimited file with chromosome sizes"
+        }
+        exclude_list: "BED file with regions to exclude from analysis"
+        prefix: "Prefix for the output BAM"
+        contact_stats: "Contact stats file"
+        merged_read_one_mapping_stats: "Mapping stats for read one from HiC-Pro"
+        merged_read_two_mapping_stats: "Mapping stats for read two from HiC-Pro"
+        merged_pairing_stats: "Pairing stats for merged reads from HiC-Pro"
+        combined_bams: "Array of BAM files to merge"
+    }
+
+    input {
+        File all_valid_pairs
+        File chromsizes
+        File contact_stats
+        File merged_read_one_mapping_stats
+        File merged_read_two_mapping_stats
+        File merged_pairing_stats
+        Array[File] combined_bams
+        String prefix
+        File? exclude_list
+    }
+
+    call hilow.converthic { input:
+        all_valid_pairs,
+        chromsizes,
+    }
+
+    if (defined(exclude_list)) {
+        call hilow.filter { input:
+            all_valid_pairs,
+            chromsizes,
+            exclude_list = select_first([exclude_list, ""]),
+        }
+    }
+
+    call hilow.qcreport { input:
+        prefix,
+        all_valid_pairs_stats = contact_stats,
+        mapping_stats_read1 = merged_read_one_mapping_stats,
+        mapping_stats_read2 = merged_read_two_mapping_stats,
+        pairing_stats = merged_pairing_stats,
+    }
+
+    call samtools.merge { input:
+        bams = combined_bams,
+        prefix = prefix + ".merged",
+        attach_rg = false,
+    }
+
+    call samtools.fixmate { input:
+        bam = merge.merged_bam,
+    }
+
+    call picard.sort as query_sort { input:
+        bam = fixmate.fixmate_bam,
+        sort_order = "queryname",
+        prefix = prefix + ".querysorted",
+    }
+
+    call picard.mark_duplicates { input:
+        bam = query_sort.sorted_bam,
+        prefix = prefix + ".markduplicates",
+    }
+
+    call picard.sort as coordinate_sort { input:
+        bam = select_first([mark_duplicates.duplicate_marked_bam, ""]),
+        sort_order = "coordinate",
+        prefix = prefix + ".sorted",
+    }
+
+    call samtools.index { input:
+        bam = coordinate_sort.sorted_bam,
+    }
+
+    output {
```

Comment:
A lot of intermediate BAMs created. Do we want to expose them using an `IntermediateFiles?` struct output? Maybe overkill.

Reply:
I'm inclined to say no. I don't think there is much use for the intermediate files. The only exception is the query-sorted BAM, but I'm not sure that's worth adding another (large) output file.

Reply:
Fair enough.